compiler/crates/graphql-text-printer/src/print_to_text.rs

```diff
@@ -638,17 +596,99 @@ impl<'schema, 'writer, W: Write> Printer<'schema, 'writer, W> {
-                write!(self.writer, "{}: ", alias.item)?;
+                write!(self.writer, "{}", alias.item)?;
+                self.colon_separator()?;
             }
         }
         write!(self.writer, "{}", name)
     }
 
-    fn next_line(&mut self) -> FmtResult {
+    fn print_argument(&mut self, argument: &Argument) -> FmtResult {
+        write!(self.writer, "{}", argument.name.item)?;
+        self.colon_separator()?;
+        self.print_value(&argument.value.item)
+    }
+
+    fn print_type_condition(&mut self, type_condition: Type) -> FmtResult {
+        self.text("on")?;
+        self.space(false)?;
+        write!(self.writer, "{}", self.schema.get_type_name(type_condition))
+    }
+
+    fn print_variable_definition(&mut self, var_def: &VariableDefinition) -> FmtResult {
+        self.print_variable(&var_def.name.item)?;
+        self.colon_separator()?;
+        self.print_type(&var_def.type_)?;
+
+        if let Some(default_value) = &var_def.default_value {
+            self.print_default_value(&default_value.item)?;
+        }
+
+        self.print_directives(&var_def.directives, None, None)
+    }
+
+    fn print_object_field(&mut self, arg: &ConstantArgument) -> FmtResult {
+        if self.options.json_format {
+            write!(self.writer, "\"{}\"", arg.name.item)?;
+        } else {
+            write!(self.writer, "{}", arg.name.item)?;
+        }
+        self.colon_separator()?;
+        self.print_constant_value(&arg.value.item)
+    }
+
+    fn print_variable(&mut self, variable: &StringKey) -> FmtResult {
+        write!(self.writer, "${}", variable)
+    }
+
+    fn print_default_value(&mut self, value: &ConstantValue) -> FmtResult {
+        self.space(true)?;
+        self.text("=")?;
+        self.space(true)?;
+        self.print_constant_value(&value)
+    }
+
+    fn print_type(&mut self, type_: &TypeReference) -> FmtResult {
+        write!(self.writer, "{}", self.schema.get_type_string(&type_))
+    }
+
+    fn next_line(&mut self, can_collapse: bool) -> FmtResult {
+        if self.options.compact && can_collapse {
+            return Ok(());
+        }
         writeln!(self.writer)?;
         for _ in 0..self.indentation {
             write!(self.writer, "  ")?;
         }
         Ok(())
     }
+
+    fn text(&mut self, text: &str) -> FmtResult {
+        write!(self.writer, "{}", text)
+    }
+
+    fn space(&mut self, can_collapse: bool) -> FmtResult {
+        if self.options.compact && can_collapse {
+            return Ok(());
+        }
+        write!(self.writer, " ")
+    }
+
+    fn comma_separator(&mut self) -> FmtResult {
+        self.text(",")?;
+        self.space(true)
+    }
+
+    fn colon_separator(&mut self) -> FmtResult {
+        self.text(":")?;
+        self.space(true)
+    }
+
+    fn item_separator(&mut self) -> FmtResult {
+        if self.options.compact {
+            write!(self.writer, ",")
+        } else {
+            self.next_line(false)
+        }
+    }
```

Comment:
These separator utils are the main change here, they have conditional behaviour in them which avoids the need for `self.options.compact` checks everywhere